packages/design-system/src/components/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,168 @@
+
+    this.state = {
+      searchTerm: '',
+      isInCompactMode: this.props.compact
+    };
+  }
+
+  handleChange = e => {
+    const { value } = e.target;
+    const { debounceInMs, onChange } = this.props;
+
+    this.setState({
+      searchTerm: value
+    });
+
+    if (debounceInMs) {
+      this.onChangeDebounced(value);
+    } else {
+      onChange(value);
+    }
+  };
+
+  handleKeyPress = key => {
+    const { onSubmit } = this.props;
+
+    if (onSubmit) {
+      if (key.keyCode === KeyCodes.enter) {
+        onSubmit(this.state.searchTerm);
+      }
+    }
+  };
+
+  handleClear = () => {
+    this.setState({
+      searchTerm: ''
+    });
+
+    this.inputRef.current.focus();
+  };
+
+  handleKeyDown = key => {
+    if (key.keyCode === KeyCodes.enter) {
+      this.handleClear();
+    }
+  };
+
+  toggleCompactMode = () => {
+    this.setState(state => ({
+      isInCompactMode: !state.isInCompactMode
+    }));
+  };
+
+  render() {
+    const {
+      className,
+      placeholder,
+      value,
+      loading,
+      compact,
+      error,
+      onChange,
+      onSubmit,
+      ...restProps
+    } = this.props;
+
+    const { searchTerm, isInCompactMode } = this.state;
+
+    const shouldDisplayCloseButton = searchTerm && !loading && !isInCompactMode;
+
+    return (
+      <span className={className}>
+        <div className={styles[`${baseClass}__container`]}>
+          <SearchIcon
+            fill="#424d57"
+            width="18px"
+            height="18px"
+            className={cx(`lc-${baseClass}__icon--search`, {
+              [`lc-${baseClass}__icon--search-compact`]: compact
+            })}
+            onClick={compact ? this.toggleCompactMode : null}
+          />
+          {loading && (
+            <Loader
+              size="small"
+              className={styles[`${baseClass}__icon--loader`]}
+            />
+          )}
+          <input
+            type="input"
+            placeholder={placeholder}
+            ref={this.inputRef}
+            value={value || this.state.searchTerm}
+            onInput={this.handleChange}
+            onKeyDown={this.handleKeyPress}
+            className={cx(`lc-${baseClass}__input`, {
+              [`lc-${baseClass}__input-compact`]: isInCompactMode
+            })}
+            {...restProps}
+          />
+          {shouldDisplayCloseButton && (
+            <CloseIcon
+              fill="#424d57"
+              width="18px"
+              height="18px"
+              onClick={this.handleClear}
+              onKeyDown={this.handleKeyDown}
+              className={styles[`${baseClass}__icon--close`]}
+              tabIndex="0"
+            />
+          )}
+        </div>
+        <span className={styles[`${baseClass}__error`]}>{error}</span>
+      </span>
+    );
+  }
+}
+
+const basePropTypes = {
+  className: PropTypes.string,
+  placeholder: PropTypes.string,
+  value: PropTypes.string,
+  loading: PropTypes.bool,
+  compact: PropTypes.bool,
```

Comment:
```suggestion
  collapsed: PropTypes.bool,
```
I see in the design that search bar can be compact which means that the input height is smaller etc. Can we add `size` property? (`size?: 'compact'`, something more in the future is possible)